xls/netlist/netlist_parser_test.cc

```diff
@@ -176,6 +201,46 @@ endmodule)";
   }
 }
 
+TEST(NetlistParserTest, Lut) {
+  // Test that a "real world" netlist parses
+  std::string netlist =
+      R"(/* Generated by Yosys 0.9+3675 (git sha1 71ca9a82, gcc 10.2.0-9 -fPIC -Os) */
+
+(* top =  1  *)
+(* src = "and.v:1.1-3.10" *)
+module main(a0, a1, a2, a3, q0);
+  (* src = "and.v:1.19-1.21" *)
+  input a0;
+  (* src = "and.v:1.29-1.31" *)
+  input a1;
+  (* src = "and.v:1.39-1.41" *)
+  input a2;
+  (* src = "and.v:1.49-1.51" *)
+  input a3;
+  (* src = "and.v:1.60-1.62" *)
+  output q0;
+  (* module_not_derived = 32'd1 *)
+  (* src = "/usr/local/google/home/fcz/opt/yosys/bin/../share/yosys/ice40/cells_map.v:26.33-27.52" *)
+  SB_LUT4 #(
+    .LUT_INIT(16'h8000)
+  ) q0_SB_LUT4_O (
+    .I0(a0),
+    .I1(a1),
+    .I2(a2),
+    .I3(a3),
+    .O(q0)
+  );
+endmodule
+)";
+  Scanner scanner(netlist);
+  XLS_ASSERT_OK_AND_ASSIGN(CellLibrary cell_library, MakeFakeCellLibrary());
+  XLS_ASSERT_OK_AND_ASSIGN(std::unique_ptr<Netlist> n,
+                           Parser::ParseNetlist(&cell_library, &scanner));
+  XLS_ASSERT_OK_AND_ASSIGN(const Module* m, n->GetModule("main"));
+  EXPECT_EQ("main", m->name());
+  XLS_ASSERT_OK(m->ResolveCell("q0_SB_LUT4_O"));
```

Comment:
Anyway to check the 0x8000 value?

Reply:
The mask value gets discarded during parsing. I added a check for the cell library entry name (which contains the mask). A more thorough check would be to compare the `StateTable` against an expected value but this would require defining `StateTable::operator==`